### ISSUE TITLE
Java runtime compatibility

### ISSUE THREAD

**Issue by Jeffwan** [oap-project/raydp#131]
I am playing raydp inside head pod on k8s. 

```
import ray
import raydp

# connect to the cluster
ray.init(address='auto')

2021-04-08 15:26:01,060	INFO worker.py:654 -- Connecting to existing Ray cluster at address: 192.168.13.21:6379
{'node_ip_address': '192.168.13.21', 'raylet_ip_address': '192.168.13.21', 'redis_address': '192.168.13.21:6379', 'object_store_address': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/plasma_store', 'raylet_socket_name': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/raylet', 'webui_url': '0.0.0.0:8265', 'session_dir': '/tmp/ray/session_2021-04-08_15-02-34_751440_101', 'metrics_export_port': 49804, 'node_id': 'adb6bbd4e36d880a70ee156fd05895c0b8a4bad92e88f14a733b035c'}

ray.nodes()

[{'NodeID': 'bbf00a25f2fc5fecb28245c63b57d18700e5819532233dbc0bc1d936', 'Alive': True, 'NodeManagerAddress': '192.168.13.22', 'NodeManagerHostname': 'example-cluster-ray-worker-jr8js', 'NodeManagerPort': 41061, 'ObjectManagerPort': 43811, 'ObjectStoreSocketName': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/plasma_store', 'RayletSocketName': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/raylet', 'MetricsExportPort': 50616, 'alive': True, 'Resources': {'memory': 3006477107.0, 'CPU': 2.0, 'object_store_memory': 1273641369.0, 'bar': 1.0, 'foo': 1.0, 'node:192.168.13.22': 1.0}}, {'NodeID': 'adb6bbd4e36d880a70ee156fd05895c0b8a4bad92e88f14a733b035c', 'Alive': True, 'NodeManagerAddress': '192.168.13.21', 'NodeManagerHostname': 'example-cluster-ray-head-xnk64', 'NodeManagerPort': 35547, 'ObjectManagerPort': 45743, 'ObjectStoreSocketName': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/plasma_store', 'RayletSocketName': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/raylet', 'MetricsExportPort': 49804, 'alive': True, 'Resources': {'memory': 3006477107.0, 'node:192.168.13.21': 1.0, 'CPU': 2.0, 'object_store_memory': 1273423872.0}}, {'NodeID': '02df037ac63f257c289e1d16762cadff26c9f8b0e5714738416c09f9', 'Alive': True, 'NodeManagerAddress': '192.168.13.23', 'NodeManagerHostname': 'example-cluster-ray-worker-5rknz', 'NodeManagerPort': 42665, 'ObjectManagerPort': 36957, 'ObjectStoreSocketName': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/plasma_store', 'RayletSocketName': '/tmp/ray/session_2021-04-08_15-02-34_751440_101/sockets/raylet', 'MetricsExportPort': 64861, 'alive': True, 'Resources': {'memory': 3006477107.0, 'object_store_memory': 1273514803.0, 'foo': 1.0, 'node:192.168.13.23': 1.0, 'bar': 1.0, 'CPU': 2.0}}]

```

Ray environment is correct, however, it encounter some issues when I use raydp. 
**Seems Java run time is not compatible with current ray**. 

Java Version:  openjdk 11.0.10 2021-01-19
Ray Cluster: 2.0.0-dev
Raydp: 0.2.0

```

spark = raydp.init_spark("K8S test", num_executors=2, executor_cores=1, executor_memory="512M")


Setting default log level to "WARN".
To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
.....
2021-04-08 15:27:23 ERROR Inbox:94 - An error happened while processing message in the inbox for RAY_APP_MASTER
java.lang.NoSuchMethodError: 'io.ray.api.call.ActorCreator io.ray.api.call.ActorCreator.setJvmOptions(java.lang.String)'
	at org.apache.spark.raydp.AppMasterJavaUtils.createExecutorActor(AppMasterJavaUtils.java:47)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.org$apache$spark$deploy$raydp$RayAppMaster$RayAppMasterEndpoint$$requestNewExecutor(RayAppMaster.scala:190)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.$anonfun$schedule$1(RayAppMaster.scala:177)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.org$apache$spark$deploy$raydp$RayAppMaster$RayAppMasterEndpoint$$schedule(RayAppMaster.scala:176)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint$$anonfun$receive$1.applyOrElse(RayAppMaster.scala:104)
	at org.apache.spark.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:115)
	at org.apache.spark.rpc.netty.Inbox.safelyCall(Inbox.scala:213)
	at org.apache.spark.rpc.netty.Inbox.process(Inbox.scala:100)
	at org.apache.spark.rpc.netty.MessageLoop.org$apache$spark$rpc$netty$MessageLoop$$receiveLoop(MessageLoop.scala:75)
	at org.apache.spark.rpc.netty.MessageLoop$$anon$1.run(MessageLoop.scala:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Exception in thread "dispatcher-event-loop-1" java.lang.NoSuchMethodError: 'io.ray.api.call.ActorCreator io.ray.api.call.ActorCreator.setJvmOptions(java.lang.String)'
	at org.apache.spark.raydp.AppMasterJavaUtils.createExecutorActor(AppMasterJavaUtils.java:47)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.org$apache$spark$deploy$raydp$RayAppMaster$RayAppMasterEndpoint$$requestNewExecutor(RayAppMaster.scala:190)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.$anonfun$schedule$1(RayAppMaster.scala:177)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.org$apache$spark$deploy$raydp$RayAppMaster$RayAppMasterEndpoint$$schedule(RayAppMaster.scala:176)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint$$anonfun$receive$1.applyOrElse(RayAppMaster.scala:104)
	at org.apache.spark.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:115)
	at org.apache.spark.rpc.netty.Inbox.safelyCall(Inbox.scala:213)
	at org.apache.spark.rpc.netty.Inbox.process(Inbox.scala:100)
	at org.apache.spark.rpc.netty.MessageLoop.org$apache$spark$rpc$netty$MessageLoop$$receiveLoop(MessageLoop.scala:75)
	at org.apache.spark.rpc.netty.MessageLoop$$anon$1.run(MessageLoop.scala:41)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

action doesn't get any resources. It keeps waiting here. I can confirm that my cluster has enough resources. I think this is still related to spark session initialization

```
>>> spark.range(0, 100).count()
21/04/08 15:28:52 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
21/04/08 15:29:07 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
21/04/08 15:29:22 WARN TaskSchedulerImpl: Initial job has not accepted any resources; check your cluster UI to ensure that workers are registered and have sufficient resources
...
```

**Comment by ConeyLiu** [comment]
Hi @kira-lin, could you take a look at this?

**Comment by kira-lin** [comment]
Hi @Jeffwan , you are using Ray 2.0.0-dev, but as you see raydp is not compatible with it. `setJvmOptions` is changed to accept List of String instead of String in 2.0.0-dev, that's why you are seeing this exception. Therefore raydp failed to start executors, and the spark job hangs. 
Is it possible to use Ray 1.2.0 for your cluster? That is the simplest way to solve this issue.
Thanks

**Comment by Jeffwan** [comment]
@kira-lin 
Thanks for the heads up. I updated my stack to 1.2.0 and it works fine. You save my day! :D

**Comment by Jeffwan** [comment]
@kira-lin Is there a master version wheel available with ray 2.0.0-dev?  I found 1.2.0 is problematic and I really like to use latest version to be compatible with other frameworks.

**Comment by kira-lin** [comment]
Unfortunately no, we do not support 2.0.0-dev for now, since it's not a stable release. We'll consider add support for it soon.

**Comment by kira-lin** [comment]
Hi @Jeffwan , I have created a new branch which supports Ray nightly. You can try it if you still want to use 2.0.0.

**Comment by ssiegel95** [comment]
I'm using a ray container (rayproject/ray:1.4.0-py37) and I see the same error. Is there a workaround available for v1.4?

```
21/07/16 21:01:34 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Exception in thread "dispatcher-event-loop-1" java.lang.NoSuchMethodError: io.ray.api.call.ActorCreator.setJvmOptions(Ljava/lang/String;)Lio/ray/api/call/ActorCreator;
	at org.apache.spark.raydp.RayExecutorUtils.createExecutorActor(RayExecutorUtils.java:45)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.org$apache$spark$deploy$raydp$RayAppMaster$RayAppMasterEndpoint$$requestNewExecutor(RayAppMaster.scala:227)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.$anonfun$schedule$1(RayAppMaster.scala:214)
	at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:158)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint.org$apache$spark$deploy$raydp$RayAppMaster$RayAppMasterEndpoint$$schedule(RayAppMaster.scala:213)
	at org.apache.spark.deploy.raydp.RayAppMaster$RayAppMasterEndpoint$$anonfun$receive$1.applyOrElse(RayAppMaster.scala:121)
	at org.apache.spark.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:115)
	at org.apache.spark.rpc.netty.Inbox.safelyCall(Inbox.scala:203)
	at org.apache.spark.rpc.netty.Inbox.process(Inbox.scala:100)
	at org.apache.spark.rpc.netty.MessageLoop.org$apache$spark$rpc$netty$MessageLoop$$receiveLoop(MessageLoop.scala:75)
	at org.apache.spark.rpc.netty.MessageLoop$$anon$1.run(MessageLoop.scala:41)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by kira-lin** [comment]
We have not released new version yet, so the support for Ray 1.4.0 is in raydp-nightly. @ssiegel95 Sorry for inconvenience.
